pisa/analysis/analysis.py

```diff
@@ -986,11 +992,14 @@ def _fit_octants(self, data_dist, hypo_maker, metric, external_priors_penalty,
             if not self.blindness:
                 logging.info('Accepting initial-octant fit')
         
-        hypo_maker.update_params(best_fit_info.params)
-        # We take the original angle with the original range and nominal value
-        # and only adjust its fit value.
-        ang_orig.value = best_fit_info.params[angle_name].value
+        # Put the original angle parameter back into the hypo maker
         hypo_maker.update_params(ang_orig)
+
+        # Copy the fitted parameter values from the best fit case into the hypo maker's parameter values
+        # Also reinstate the original parameter range for the angle
+        for best_fit_param in best_fit_info.params.free :
+            hypo_maker.params[best_fit_param.name].value = best_fit_param.value
```

Comment:
I know this was already merged, but I missed it: The parameters in a HypoMaker cannot be set like this! The `hypo_maker.params` property is always a copy that can't be manipulated, and trying to do it gives strange errors later on! We should really try to do something about this, because it's a mistake that happens all the time. 🙈Maybe we could have a property `read_only` for parameters so that attempting to write values like this will immediately raise an exception.